migration/migration.go

```diff
@@ -464,6 +465,29 @@ func calculateBlockCommitments(txn db.Transaction, network *utils.Network) error
 	return workerPool.Wait()
 }
 
+func calculateL1MsgHashes(txn db.Transaction, n *utils.Network) error {
```

Comment:
few ideas how to improve performance:
1. remove mutex
2. use fixed pool of goroutines with `workerPool.Go()` (if it's used in a loop then it constantly using some resources for creating new goroutine and shutting it down)

something like this:
```
workerPool := ...

head, err := blockchain.Head()
if err != nil {
    ... // in case ErrKeyNotFound - return nil
}
blocks := make(chan uint64, 1024) // actual capacity of buffered channel is actually not super important
go func() {
    for block := range head.Number {
         blocks <- block
    }
    close(blocks)
}()
for range numOfWorkers {
    workerPool.Go(func() error {
        for blockNumber := range blocks {
            ... := blockchain.BlockByNumber(txn, blockNumber)
            ... := blockchain.StoreL1HandlerMsgHashes(txn, block)
        }
    })
}
```

Reply:
Also we can significantly improve it if we fetch only necessary data (i.e. transactions). For that we need:
1. receive list of transactions in `StoreL1HandlerMsgHashes` instead of whole block
2. make `transactionsByBlockNumber` public function

Reply:
I wonder if we could also apply these changes to `calculateBlockCommitments`?

Reply:
I updated both `calculateL1MsgHashes` and `calculateBlockCommitments`